pkg/asset/password/password.go

```diff
@@ -114,6 +114,14 @@ func (a *KubeadminPassword) Load(f asset.FileFetcher) (found bool, err error) {
 	}
 
 	a.PasswordHash = hashFile.Data
+
+	// Also load the kubeadmin-password from the tls dir if it exists,
+	passwordFilePath := filepath.Join("tls", "kubeadmin-password")
+	passwordFile, err := f.FetchByName(passwordFilePath)
+	if err == nil {
+		a.Password = string(passwordFile.Data)
+	}
+
```

Comment:
From my local tests it looks like that the password must be persisted, otherwise when the assisted-service will invoke the commands having this asset as a dependency (ie `create ignition-configs`) then they will generate an empty password in the local cluster data folder.

I' suggest to modify the below code as: 
```
	// Assisted-service expects to always see a password file, so generate an
	// empty one
	a.File = &asset.File{
		Filename: kubeadminPasswordPath,
		Data:     []byte(a.Password),
	}
```

So that the file will get populated when `a.Password` is not empty (and left empty otherwise)

Reply:
Thanks for testing this. Updated.